liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/ServerFeatureSupport.java

```diff
@@ -347,4 +369,65 @@ protected Plugin getPluginForProject(String groupId, String artifactId, MavenPro
         }
         return plugin;
     }
+
+    /**
+     * Initialize the toolchain by calling the toolchain goal.
+     * If useToolchainJdk is set to true, this method will also set the toolchain variable.
+     *
+     * @throws MojoExecutionException If an exception occurred while running toolchain goal
+     */
+    protected void initToolchain() throws MojoExecutionException {
+        // Skip if toolchain support is not enabled
+        if (jdkToolchain == null || toolchainManager == null) {
+            return;
+        }
+        List<Toolchain> tcs = toolchainManager.getToolchains(session, "jdk", jdkToolchain);
+        if (tcs != null && !tcs.isEmpty()) {
+            this.toolchain = tcs.get(0);
+            getLog().info(MessageFormat.format(messages.getString("info.toolchain.initialized"), this.toolchain));
+        } else {
+            getLog().warn(MessageFormat.format(messages.getString("warn.toolchain.not.available"), jdkToolchain));
+        }
+    }
+
+    /**
+     *
+     * @param serverEnvLines lines from server.env
+     * @param jvmOptionsLines lines from jvm.options
+     * @return true or false
+     */
+    protected static boolean isJavaHomeSet(List<String> serverEnvLines, List<String> jvmOptionsLines) {
+        boolean javaHomeSet = false;
+        for (String serverEnvLine : serverEnvLines) {
+            if (serverEnvLine.startsWith("JAVA_HOME=")) {
+                javaHomeSet = true;
+                break;
+            }
+        }
+        for (String serverEnvLine : jvmOptionsLines) {
+            if (serverEnvLine.contains("JAVA_HOME=")) {
+                javaHomeSet = true;
+                break;
+            }
+        }
+        return javaHomeSet;
+    }
+
+    /**
+     * Get JDK home directory from toolchain
+     *
+     * @param toolchain The toolchain to get JDK home from
+     * @return The JDK home directory path, or null if it could not be determined
+     */
+    protected String getJdkHomeFromToolchain(Toolchain toolchain) {
+        String jdkHome = toolchain.findTool("java");
+        if (jdkHome != null) {
+            File javaFile = new File(jdkHome);
+            File binDir = javaFile.getParentFile();
+            if (binDir != null) {
+                return binDir.getParent();
```

Comment:
Do we want absolute path, or canonical path?...which one does `getParent()` give us?

Should we give some warning if javaFile does not exist, or if the parent directory does not exist? We return null below without logging anything.